### PR TITLE
Big refactor to make this repo the one source of our eslint configs

### PR DESCRIPTION
**Pull request by apellerano-pw**
Some eslint configuration has leaked into each project when it should reside here. This config package can also specify multiple configs for different environments. For now it will contain 1 for generic node projects and 1 for ember projects. We can easily add more as needed.

The end result is that each of our ALI/client/* folders can have an extremely simple .eslintrc.js like this
```javascript
module.exports = {
  root: true,
  extends: [
    '@prosperworks/eslint-config/ember',
  ],
};
```

I renamed the project to `@prosperworks/eslint-config` as this is the recommended way to name eslint-config projects inside an npm scope. I deprecated `@prosperworks/eslint-config-prosperworks` on npm for versions under 2.0.0, which this PR updates the version to.

I deleted a lot of cruft from when we forked the airbnb style related to features we aren't using like testing tools and deploy scripts. We can re-implement that functionality when our workflow has a need for it. I also deleted the eslint rules governing this project as the yo-dawg-ness of having a lint config so your linter can lint your lint config makes it difficult for newcomers to grok this project.

### CODE REVIEW

File: rules/.eslintrc [hunk 1]
```diff
@@ -1,5 +0,0 @@
-{
```
Comment (apellerano-pw) [inline rules/.eslintrc:1]: This was just a lint config for the folder, it's not part of our exported lint config. (Now do you see why I deleted them?)

File: linters/.eslintrc [hunk 2]
```diff
@@ -1,5 +0,0 @@
-// Use this file as a starting point for your project's .eslintrc.
```
Comment (apellerano-pw) [inline linters/.eslintrc:1]: ember-cli gives us a starting point with the non-deprecated .eslintrc.js, we don't need a file template

File: .eslintrc [hunk 3]
```diff
@@ -1,3 +0,0 @@
-{
```
Comment (apellerano-pw) [inline .eslintrc:1]: This was just a lint config for the folder, it's not part of our exported lint config. (Now do you see why I deleted them?)

File: package.json [hunk 4]
```diff
@@ -1,31 +1,26 @@
...
-    "eslint-find-rules": "^1.14.3",
-    "eslint-plugin-import": "^2.2.0"
+    "eslint": "^4.19.0",
+    "eslint-plugin-ember": "^5.1.0",
+    "eslint-plugin-import": "^2.11.0"
+  },
+  "peerDependencies": {
+    "eslint": "^4.19.0",
+    "eslint-plugin-ember": "^5.1.0",
+    "eslint-plugin-import": "^2.11.0"
```
Comment (apellerano-pw) [inline package.json:21]: When eslint-configs contain rules for plugins, the documentation states they must list those plugins as peer dependencies, to alert projects that they should also install those dependencies. This is stupid and lots of ppl agree, but npm maintainers haven't budged in 3 years of debate on this topic so we'll just do it their stupid way.

File: packages/eslint-config-prosperworks/index.js [hunk 5]
```diff
@@ -1,22 +0,0 @@
-/* eslint-env node */
```
Comment (apellerano-pw) [inline packages/eslint-config-prosperworks/index.js:1]: Not sure what this packages folder was doing but it's not needed for an eslint-config package. By default they read index.js in the root with additional configs also existing in the root.

File: package.json [hunk 6]
```diff
@@ -1,31 +1,26 @@
...
+  "name": "@prosperworks/eslint-config",
+  "version": "2.0.0",
   "description": "A mostly reasonable approach to JavaScript.",
-  "scripts": {
-    "prelint": "editorconfig-tools check ./rules/*",
-    "lint": "eslint ./rules",
-    "fix": "eslint ./rules --fix"
-  },
-  "repository": "git+https://git@github.com/ProsperWorks/js-style.git",
-  "homepage": "https://github.com/ProsperWorks/js-style",
```
Comment (ghost) [inline package.json:11]: do we want to get rid of this homepage link? i'm totally speaking without context, but i would think that maybe npm would parse this file to figure out how to link the page and the repo.

or is it that by doing `github:ProsperWorks/js-style` you get the same benefits?

**Review by epetrsn — APPROVED** [review]

File: ember.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
+/**
+ * Style guide enforcement based on the PW standard and addtl ember-specific rules.
```
Comment (santosh-kalidindi) [inline ember.js]: nit: typo addtl

**Review by santosh-kalidindi — APPROVED** [review]
LGTM

File: package.json (hunk 6 above)
Comment (apellerano-pw) [reply package.json:11]: the repo is linked via `repository`, see an example npm page https://www.npmjs.com/package/lodash

homepage would be for a marketing landing or doc site or something

**Review by ghost — APPROVED** [review]
yep, looks like javascript.

**Pull request merged**